### PR TITLE
kvserver: move SideloadStorage to a package

### LINKED ISSUE #91899: kvserver: factor out SideloadStorage to log package
[SideloadStorage](https://github.com/cockroachdb/cockroach/blob/47c7b3a1bc047fc3e481cf12166885b39519c022/pkg/kv/kvserver/replica_sideload.go#L30) interface and implementation is one of the pieces of a separated log storage. Within the bigger effort of factoring out the log storage, moving the `SideloadStorage` to a separate package is among reasonable first steps.

Part of #91979
Epic: CRDB-220

Jira issue: CRDB-21475

### PR DESCRIPTION
**Pull request by pav-kv**
Fixes #91899
Release note: None

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Pending #91898 merge, and figuring out where to place the `syncing_write.go` file.

**Comment by pavelkalinnikov** [comment]
Ready for review (only need to look at the last 2 commits).

File: pkg/kv/kvserver/logstore/sideload.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ type SideloadStorage interface {
 //
 // The provided slice is not modified, though the returned slice may be backed
 // in parts or entirely by the same memory.
-func maybeSideloadEntries(
+func MaybeSideloadEntries(
```
Comment (pavelkalinnikov) [inline pkg/kv/kvserver/logstore/sideload.go:75]: All these will be unexported again when more things move to `logstore`.

**Comment by pavelkalinnikov** [comment]
I don't understand the CI failure. Does it look like another circular dependency?

File: pkg/kv/kvserver/logstore/sideload.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ import (
 	"go.etcd.io/etcd/raft/v3/raftpb"
 )
 
-var errSideloadedFileNotFound = errors.New("sideloaded file not found")
+var ErrSideloadedFileNotFound = errors.New("sideloaded file not found")
```
Comment (tbg) [inline pkg/kv/kvserver/logstore/sideload.go:26]: Doesn't this need a comment or linters will complain?

**Review by tbg — APPROVED** [review]

File: pkg/kv/kvserver/logstore/sideload.go (hunk 2 above)
Comment (tbg) [reply pkg/kv/kvserver/logstore/sideload.go:26]: Huh, apparently not.

Comment (pavelkalinnikov) [reply pkg/kv/kvserver/logstore/sideload.go:26]: Yeah, maybe this lint is disabled. I'm unexporting this though, it's used only inside the package. Will now revisit/fix the exporting/commenting in this PR, and then merge.

**Comment by pavelkalinnikov** [comment]
bors r=tbg

**Pull request merged**